sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -452,22 +455,173 @@ def fetch_all_values(cls, cursor, counters_list, logger):
 
     def fetch_metric(self, rows, columns):
         database_name = columns.index("name")
+        db_state_desc_index = columns.index("state_desc")
         value_column_index = columns.index(self.column)
 
         for row in rows:
             if row[database_name] != self.instance:
                 continue
 
             column_val = row[value_column_index]
-
+            db_state_desc = row[db_state_desc_index]
             metric_tags = [
                 'database:{}'.format(str(self.instance)),
+                'database_state_desc:{}'.format(str(db_state_desc)),
```

Comment:
`database_state_desc` isn't related to AlwaysOn metrics, but is just a small tag addition to `SqlDatabaseStats` metrics class. I can move this to its own PR to make it more clear.

Reply:
Nice improvement, no need for a separate PR.